pkg/oc/cli/cmd/rollout/latest.go

```diff
@@ -175,17 +238,70 @@ func (o RolloutLatestOptions) RunRolloutLatest() error {
 		}
 
 		if err != nil {
-			return err
+			return nil, err
 		}
+	}
+
+	return dc, nil
+}
 
+// RunRolloutLatest runs the latest rollouts.
+func (o RolloutLatestOptions) RunRolloutLatest() error {
+	var revision string
+	info := o.infos[0]
+	switch obj := info.Object.(type) {
+	case *deployapi.DeploymentConfig:
+		dc, err := o.rolloutDeploymentConfig(obj)
+		if err != nil {
+			return err
+		}
+		revision = fmt.Sprintf("%d", dc.Status.LatestVersion)
 		info.Refresh(dc, true)
+	case *kextensions.Deployment:
+		updatedObj, err := o.rolloutFromAnnotation(obj, func(in runtime.Object) (runtime.Object, error) {
+			return o.kc.Extensions().Deployments(obj.Namespace).Update(in.(*kextensions.Deployment))
+		})
+		if err != nil {
+			return fmt.Errorf("%s %v", info.Mapping.Resource, err)
+		}
+		currentRevision, _ := util.Revision(updatedObj)
+		revision = fmt.Sprintf("%d", currentRevision)
+		info.Refresh(updatedObj, true)
+	case *kextensions.DaemonSet:
+		updatedObj, err := o.rolloutFromAnnotation(obj, func(in runtime.Object) (runtime.Object, error) {
+			return o.kc.Extensions().DaemonSets(obj.Namespace).Update(in.(*kextensions.DaemonSet))
+		})
+		if err != nil {
+			return fmt.Errorf("%s %v", info.Mapping.Resource, err)
+		}
+		// TODO: Does DaemonSets have revision?
+		info.Refresh(updatedObj, true)
+	case *kapps.StatefulSet:
+		updatedObj, err := o.rolloutFromAnnotation(obj, func(in runtime.Object) (runtime.Object, error) {
+			return o.kc.Apps().StatefulSets(obj.Namespace).Update(in.(*kapps.StatefulSet))
+		})
+		if err != nil {
+			return fmt.Errorf("%s %v", info.Mapping.Resource, err)
+		}
+		s := updatedObj.(*kapps.StatefulSet)
+		revision = s.Status.CurrentRevision
+		info.Refresh(updatedObj, true)
+	case *kbatch.CronJob:
+		updatedObj, err := o.rolloutFromAnnotation(obj, func(in runtime.Object) (runtime.Object, error) {
+			return o.kc.Batch().CronJobs(obj.Namespace).Update(in.(*kbatch.CronJob))
+		})
+		if err != nil {
+			return fmt.Errorf("%s %v", info.Mapping.Resource, err)
+		}
+		info.Refresh(updatedObj, true)
+	default:
+		return fmt.Errorf("manual rollouts are not supported for %s", info.Mapping.Resource)
 	}
 
-	if o.output == "revision" {
-		fmt.Fprintf(o.out, fmt.Sprintf("%d", dc.Status.LatestVersion))
+	if o.output == "revision" && len(revision) > 0 {
```

Comment:
Having output set to "revision" and not printing anything if empty might be confusing since user explicitly stated he wants to print that - I think I saw some of our commands print <no-value> or something like that. Or an error when specifying this flag with DaemonSets.

Reply:
if we don't have revision we just print normal message

Reply:
Well, that's exactly what bothers me. User explicitly used `--output=revision` so it should be either `<no-value>` or an error. 
We shouldn't just print success and ignore it.